### PR TITLE
homogenization - parallel computation using MPI 

### PR DESCRIPTION
**Pull request by vlukes**

### CODE REVIEW

**Comment by rc** [comment]
Thanks, LGTM!

Only some minor points to fix:

- update the docstrings (`!!!`) of `HomogenizationWorker`, `HomogenizationWorkerMulti`, `HomogenizationWorkerMultiMPI`
- in `HomogenizationEngine.call()`: mlutiprocessing -> multiprocessing

**Comment by vlukes** [comment]
Fixed.

File: sfepy/homogenization/engine.py [hunk 1]
```diff
@@ -67,16 +62,43 @@ class HomogenizationWorker(object):
...
+        post_process_hook : function
+            The postprocessing hook.
+        req_info : dict
+            The definition of correctors.
+        coef_info : dict
+            The definition of homogenized coefficients.
+        micro_coors : array
+            The configurations of multiple microstructures.
+        store_micro_idxs : list of int
+            The indicies of microstructures which results are to be stored.
```
Comment (rc) [inline sfepy/homogenization/engine.py]: The indices of microstructures whose results are to be stored.

File: sfepy/homogenization/engine.py [hunk 2]
```diff
@@ -67,16 +62,43 @@ class HomogenizationWorker(object):
...
+        time_tag : str
+            The label corresponding to the actual time step and iteration,
+            used in the corrector file names.
+
+        Returns
+        -------
+        dependencies : dict
+            It contains computed correctors and coefficients.
```
Comment (rc) [inline sfepy/homogenization/engine.py]: The computed correctors and coefficients.

File: sfepy/homogenization/engine.py [hunk 3]
```diff
@@ -287,7 +309,27 @@ def __init__(self, num_workers):
     def __call__(self, problem, options, post_process_hook,
                  req_info, coef_info,
                  micro_coors, store_micro_idxs, chunk_size, time_tag=''):
-        """ !!! """
+        """Calculate homogenized correctors and coefficients.
+
+        Parameters
+        ----------
+        The same parameters as HomogenizationWorker, extended by:
```
Comment (rc) [inline sfepy/homogenization/engine.py]: The same parameters as :class:`HomogenizationWorker`, extended by:

- please verify, that the link is correct in the html docs.

File: sfepy/homogenization/engine.py [hunk 4]
```diff
@@ -504,6 +543,88 @@ def dechunk_reqs_coefs(deps, num_chunks):
...
 
+class HomogenizationWorkerMultiMPI(HomogenizationWorkerMulti):
+    def __call__(self, problem, options, post_process_hook,
+                 req_info, coef_info,
+                 micro_coors, store_micro_idxs, chunk_size, time_tag=''):
+        """Calculate homogenized correctors and coefficients.
+
+        Parameters and Returns
+        ----------------------
+        The same parameters and returns as HomogenizationWorkerMulti.
```
Comment (rc) [inline sfepy/homogenization/engine.py]: The same parameters and returns as :class:`HomogenizationWorkerMulti`.

...

**Comment by rc** [comment]
Thanks! I will merge after you fix the small details in the comments above.

**Pull request merged**